agent/consul/leader_connect_ca.go

```diff
@@ -220,27 +220,18 @@ func (c *CAManager) initializeCAConfig() (*structs.CAConfiguration, error) {
 	if err != nil {
 		return nil, err
 	}
-	if config == nil {
-		config = c.serverConf.CAConfig
+	if config != nil {
+		return config, nil
+	}
 
-		if c.serverConf.Datacenter == c.serverConf.PrimaryDatacenter && config.ClusterID == "" {
-			id, err := uuid.GenerateUUID()
-			if err != nil {
-				return nil, err
-			}
-			config.ClusterID = id
-		}
-	} else if _, ok := config.Config["IntermediateCertTTL"]; !ok {
-		dup := *config
-		copied := make(map[string]interface{})
-		for k, v := range dup.Config {
-			copied[k] = v
+	config = c.serverConf.CAConfig
+
+	if c.serverConf.Datacenter == c.serverConf.PrimaryDatacenter && config.ClusterID == "" {
+		id, err := uuid.GenerateUUID()
+		if err != nil {
+			return nil, err
 		}
-		copied["IntermediateCertTTL"] = connect.DefaultIntermediateCertTTL.String()
```

Comment:
Where is this default-on-read behavior happening now?

Reply:
Good question! The test should show that it does work (even when the previous config came from a version that did not set a default), but I didn't explain why it worked.  

The defaulting happens in [defaultCommonConfig](https://github.com/hashicorp/consul/blob/v1.11.4/agent/connect/ca/provider_consul_config.go#L49-L57). Every provider is setup using `Provider.Configure`, and each of them calls the appropriate `Parse<Provider>CAConfig` function. And all the parse functions use `defaultCommonConfig` to populate the target of the decode. So those values are used as defaults and remain after the decode unless something overwrites them.

It looks like when this was added in #7213 we did not default it that way. The changes in #8646 removed the need for this line.

Since the default is set before in-process while constructing the provider it doesn't need to be persisted to raft. Even if the config came from an earlier version, the defaults will be set correctly.